fml/message_loop_task_queues.cc

```diff
@@ -52,39 +58,47 @@ void MessageLoopTaskQueues::RegisterTask(TaskQueueId queue_id,
   std::scoped_lock lock(GetMutex(queue_id, MutexType::kTasks));
   size_t order = order_++;
   delayed_tasks_[queue_id].push({order, std::move(task), target_time});
-  WakeUp(queue_id, delayed_tasks_[queue_id].top().GetTargetTime());
+  TaskQueueId loop_to_wake = queue_id;
+  if (subsumed_to_owner_[queue_id] != _kUnmerged) {
+    loop_to_wake = subsumed_to_owner_[queue_id];
+  }
+  WakeUp(loop_to_wake, delayed_tasks_[queue_id].top().GetTargetTime());
 }
 
 bool MessageLoopTaskQueues::HasPendingTasks(TaskQueueId queue_id) {
-  std::scoped_lock lock(GetMutex(queue_id, MutexType::kTasks));
-  return !delayed_tasks_[queue_id].empty();
+  MergedQueuesRunner merged_tasks = MergedQueuesRunner(this, queue_id);
+  return HasPendingTasksUnlocked(queue_id);
 }
 
 void MessageLoopTaskQueues::GetTasksToRunNow(
     TaskQueueId queue_id,
     FlushType type,
     std::vector<fml::closure>& invocations) {
-  std::scoped_lock lock(GetMutex(queue_id, MutexType::kTasks));
+  MergedQueuesRunner merged_tasks = MergedQueuesRunner(this, queue_id);
+
+  if (!HasPendingTasksUnlocked(queue_id)) {
+    return;
+  }
 
   const auto now = fml::TimePoint::Now();
-  DelayedTaskQueue& tasks = delayed_tasks_[queue_id];
 
-  while (!tasks.empty()) {
-    const auto& top = tasks.top();
+  while (HasPendingTasksUnlocked(queue_id)) {
+    TaskQueueId top_queue;
+    const auto& top = PeekNextTaskUnlocked(queue_id, top_queue);
     if (top.GetTargetTime() > now) {
       break;
     }
     invocations.emplace_back(std::move(top.GetTask()));
-    tasks.pop();
+    delayed_tasks_[top_queue].pop();
     if (type == FlushType::kSingle) {
```

Comment:
We no longer make use of this in the concurrent tasks queue. So lets just get rid of flush type with the default being multiple.

Reply:
Created this https://github.com/flutter/flutter/issues/36083, will follow this up.